sdk/core/src/error/azure_core_errors.rs

```diff
@@ -0,0 +1,44 @@
+use super::{Error, ErrorKind};
+
+impl From<crate::errors::Error> for Error {
+    fn from(err: crate::errors::Error) -> Error {
+        match err {
+            crate::errors::Error::Json(e) => e.into(),
+            crate::errors::Error::Header(e) => Error::new(ErrorKind::DataConversion, e),
+            crate::errors::Error::Parse(e) => Error::new(ErrorKind::DataConversion, e),
+            crate::errors::Error::HeadersNotFound(hs) => Error::with_message(
+                ErrorKind::DataConversion,
+                format!("headers not found: {}", hs.join(", ")),
+            ),
+            crate::errors::Error::HeaderNotFound(h) => Error::with_message(
+                ErrorKind::DataConversion,
+                format!("header not found: {}", h),
+            ),
+            crate::errors::Error::Http(e) => e.into(),
+            crate::errors::Error::Stream(e) => Error::new(ErrorKind::Io, e),
+            crate::errors::Error::GetToken(e) => Error::new(ErrorKind::Credential, e),
+            crate::errors::Error::HttpPrepare(e) => e.into(),
+            crate::errors::Error::Other(e) => Error::new(ErrorKind::Other, e),
+        }
+    }
+}
+
+impl From<crate::errors::HttpError> for Error {
```

Comment:
How does one get the headers or the raw response? At least the body is captured, but some services have extra information in the headers that, in some cases, will be necessary.

Reply:
This is just connector coder to make any code using the old style error still compiles while we move to the new error type. So indeed, the old error didn't allow access to headers, but the new error does.